import/permissions/permissionsplugin.h

```diff
@@ -0,0 +1,60 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SAILFISHOS_WEBVIEW_PERMISSIONS_PLUGIN_H
+#define SAILFISHOS_WEBVIEW_PERMISSIONS_PLUGIN_H
+
+#include <QtQml/QQmlExtensionPlugin>
+#include <QtQuick/QQuickItem>
+
+#include <QtCore/QTranslator>
+#include <QtCore/QLocale>
+
+namespace SailfishOS {
+
+namespace WebView {
+
+namespace Permissions {
+
+class SailfishOSWebViewPermissionsPlugin : public QQmlExtensionPlugin
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA(IID "org.qt-project.Qt.QQmlExtensionInterface")
+
+public:
+    void registerTypes(const char *uri);
+    void initializeEngine(QQmlEngine *engine, const char *uri);
+};
+
+// using custom translator so it gets properly removed from qApp when engine is deleted
+class AppTranslator: public QTranslator
+{
+    Q_OBJECT
+
+public:
+    AppTranslator(QObject *parent)
```

Comment:
Is the translation code necessary? There don't seem to be any translations here (or potential need in the future).